setuplib.php

```diff
@@ -41,7 +43,13 @@
 function create_certificates($saml2auth, $dn = false, $numberofdays = 3650) {
     global $SITE;
 
-    $opensslargs = array();
+    $signaturealgorithm = ssl_algorithms::get_default_saml_signature_algorithm();
+    if (isset($saml2auth->config->signaturealgorithm)) {
```

Comment:
Should we check if it's not empty here?

Reply:
I struggled with what checks needed to be here.

In the tests it was not being set, but I wasn't sure how it got into that state. Is there actually a way for a config variable to be empty string or not set.

Reply:
Moodle is unpredictable :) But looking at the rest of the code this line should be good.